src/config/config.server.ts

```diff
@@ -0,0 +1,220 @@
+import * as colors from 'colors';
+import * as fs from 'fs';
+import * as yaml from 'js-yaml';
+import { join } from 'path';
+
+import { AppConfig } from './app-config.interface';
+import { Config } from './config.interface';
+import { DefaultAppConfig } from './default-app-config';
+import { ServerConfig } from './server-config.interface';
+import { mergeConfig } from './config.util';
+import { isNotEmpty } from '../app/shared/empty.util';
+
+const CONFIG_PATH = join(process.cwd(), 'config');
+
+type Environment = 'production' | 'development' | 'test';
+
+const DSPACE = (key: string): string => {
+  return `DSPACE_${key}`;
+};
+
+const ENV = (key: string, prefix = false): any => {
+  return prefix ? process.env[DSPACE(key)] : process.env[key];
```

Comment:
I honestly think we should *require* the `DSPACE_` prefix.  This looks to fall back to not requiring the prefix, which means that we still may have env variable conflicts... e.g. if we ever had a field named `path` in our `config.yml`, it would try to use the `PATH` env variable _unless_ you explicitly defined `DSPACE_PATH`.

Reply:
This is not the case. This method is used to get an environment variable value. There is still the need to get an environment variable value that is not prefixed.

https://github.com/DSpace/dspace-angular/pull/1437/files#diff-0ce70abb5623f29226ce9a67ac99cfe7b4c0aa85586295984d85de2be676ac6eR35

Reply:
If you want, I can remove the use of getting NODE_ENV from this method.